astropy/io/votable/tests/converter_test.py

```diff
@@ -65,6 +66,42 @@ def test_unicode_mask():
     assert c.output("Foo", True) == ''
 
 
+def test_unicode_as_char():
+    config = {'verify': 'exception'}
+    #with catch_warnings(exceptions.W55) as w:
+    field = tree.Field(
+        None, name='unicode_in_char', datatype='char',
+        arraysize='*', config=config)
+    c = converters.get_converter(field, config=config)
+
+    c.parse("XXX")  # ASCII succeeds
+
+    with pytest.raises(
+        exceptions.W55,
+        match=r'.*FIELD \(unicode_in_char\) has datatype="char" but contains non-ASCII value.*'):
+        c.parse("zła")  # non-ASCII
+
+
+def test_unicode_as_char_binary():
+    config = {'verify': 'exception'}
+
+    field = tree.Field(
+        None, name='unicode_in_char', datatype='char',
+         arraysize='*', config=config)
+    c = converters.get_converter(field, config=config)
+    c._binoutput_var('abc', False)  # ASCII succeeds
+    with pytest.raises(exceptions.E24, match=r".*E24: Attempt to write non-ASCII value.*"):
```

Comment:
I don't think the trailing `.*` is necessary but I guess it does not hurt.

Reply:
This is interesting.  At some point I had convinced myself that was needed, but in looking at the docs now, it doesn't look like the leading `.*` is needed either.  (https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions)

I've removed the leading and trailing `.*` in 3 cases in that file and the tests still run well.